themes/foulab-2018/layouts/_default/baseof.html

```diff
@@ -1,73 +1,40 @@
-<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
-<html xmlns="http://www.w3.org/1999/xhtml">
+<!DOCTYPE html>
+<html lang="en">
   <head>
+	  <meta charset="utf-8">
 	  <title>{{ block "title" . }}{{ default .GitInfo.Subject .Title | default "Welcome to Foulab! - foulab.org" }}{{ end }}</title>
     
-    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
-  
-    <link rel="stylesheet" href="{{ .Site.BaseURL }}/css/default/default.css" />
-    <link rel="stylesheet" href="{{ .Site.BaseURL }}/css/default/app.css" />
+    <link rel="stylesheet" href="/css/default/default.css">
```

Comment:
Hmm, I'm not a Hugo expert, so I did a bit of research what's the idiomatic way to reference assets.

`{{ .Site.BaseURL }}/foo` is definitely *not* the idiomatic way, so thanks for fixing it.

I think the idiomatic way is resources.Get + r.RelPermalink, example: https://github.com/luizdepra/hugo-coder/blob/52b14a50f716cb7e69df8d0804f1b68aeec8a08c/layouts/_default/baseof.html#L41

In the short term, it doesn't give us any immediate value. But long-term:
- For JS assets, it makes it very easy to enable minification, if we want.
- For image assets, resizing
- For all asset types, resource integrity

I think it would also make it easier to serve drafts of the site on non-root URLs (eg example.com/foulab-draft/) just by changing a path in a single config file.

what do you think about switching to resources API throughout?

Maybe let's wait for @0x5c to also chime in.

Reply:
Would it work with a local deployment? A potential developer's workflow could be putting `file:///home/user/foulab-dot-org-hugo/public/index.html` into the address bar, as of right now it works pretty well with a slight config override.

Reply:
I've tried a few things and I'm basically running into different problems every time. 

The use of a variable as per the github link just doesn't work at all
```
    {{ $script := resources.Get "js/status.js" }}
    <script src="{{ $script.RelPermalink }}"></script>
```
This gives me an error about `nil pointer evaluating resource.Resource.RelPermalink`

I found alternatives for Site.BaseURL that will basically give me the same problem it did. There's something wrong in Hugo itself, because BaseURL as well as urls.absUrl just somehow don't work as expected.

So, for example, with the following code: 
```
    <link rel="stylesheet" href="{{ absLangURL "css/default/default.css" }}">
    <link rel="stylesheet" href="{{ absLangURL "css/default/app.css" }}">
```
And running `hugo serve -p 3000` I get a proper server running on `http://localhost:3000` but the HTML in the head just... don't care. 
![image](https://github.com/user-attachments/assets/b9cf2d42-f2cf-4e14-992c-7c32ad45ed4e)

Running with a `--baseURL=https://c0c01e372d48.ngrok.app/` for example, seems to work except that it doesn't, it forcefully appends the port to the end: 
![image](https://github.com/user-attachments/assets/313a584f-2405-41e8-8e5a-901a6b66e141)

And without a port specified, letting the default, I just get a default port like `https://c0c01e372d48.ngrok.app:62492/` (not sure when it decided to move away from 1313 but it sure did decide this), and I can't really test whether this would work or not since rebinding ngrok to a different port generates a new address so now I'm sort of stuck with an infinite loop of "not working".

All in all, my point here is that Hugo's system is pretty broken, and honestly, the only situation right now where this would matter would be if the baseURL somehow isn't on the root of the domain. BaseURLs really only matter when you're on some subfolder in a more complex configuration or if you're on a share hosting where you can't be on the root.

Basically, getting frustrated over trying to go around Hugo's broken system just to fix a non-problem isn't how I want to spend my days 🤣

Reply:
> Would it work with a local deployment? A potential developer's workflow could be putting `file:///home/user/foulab-dot-org-hugo/public/index.html` into the address bar, as of right now it works pretty well with a slight config override.

Using `file://` to locally develop isn't a thing we do in 2024. There are many, *many* trivial ways to start a proper thin local web server on a local machine, from using python's `SimpleHTTPServer` module, to VSCode's Live Server, to a ton of single-executable light http servers. This is a non-problem.